src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java

```diff
@@ -309,4 +310,26 @@ else if (option == WrapOption.EOL
         }
     }
 
+    /**
+     * Checks for multiline comment start and end. Also checks if there is ant code after comment.
+     *
+     * @param currentLineSubstring A string to check.
+     * @return true If it ends and there is no code after the end,
+     *     If it doesn't end on the same line
+     * */
+    private static boolean checkCodeForMultilineComment(String currentLineSubstring) {
+        boolean result = false;
+        if (currentLineSubstring.trim().indexOf("/*") == 0) {
```

Comment:
There is tree calls to `currentLineSubstring.trim()` in this method (and one in `visitToken`).
Could you please refactor the code to call it once? Also, `CommonUtil.isBlank(currentLineSubstring)` is same as `currentLineSubstring.trim().isEmpty()`. Please move all new code to this method, so `visitToken` will be something like 
```java
 if (option == WrapOption.NL
                    && !text.equals(currentLine.trim())
                    && containsNoCode(currentLineSubstring.trim()))

```
True if substring is blank;
starts with // comment;
starts with /* and ends with */
`boolean containsNoCode(String substring)`

Reply:
Done @pbludov

Reply:
@harsh-kukreja I am sorry for making you do extra work, but please pay attention to https://github.com/checkstyle/checkstyle/issues/8018#issuecomment-621872177
If it is possible to use the AST to check for violations, then we should use it.
Please investigate.

Reply:
Hey @pbludov I gave u an rply there itself please check

Reply:
@harsh-kukreja here is my AST based solution:
```
    @Override
    public void visitToken(DetailAST ast) {
        final DetailAST parent = ast.getParent();
        // we do not want to check colon for cases and defaults
        if (parent.getType() != TokenTypes.LITERAL_DEFAULT
                && parent.getType() != TokenTypes.LITERAL_CASE) {
            if (option == WrapOption.NL
                    && TokenUtil.areOnSameLine(ast, getFirstOperand(ast))
                    && !TokenUtil.areOnSameLine(ast, getSecondOperand(ast))) {
                log(ast, MSG_LINE_NEW, ast.getText());
            }
            else if (option == WrapOption.EOL && !TokenUtil.areOnSameLine(ast, getFirstOperand(ast))) {
                log(ast, MSG_LINE_PREVIOUS, ast.getText());
            }
        }
    }

    private static DetailAST getFirstOperand(DetailAST ast) {
        DetailAST result;
        if (ast.getFirstChild() != null && ast.getParent().getType() != TokenTypes.VARIABLE_DEF) {
            result = ast.getFirstChild();
            while (result.getLastChild() != null) {
                result = result.getLastChild();
            }
        }
        else {
            result = ast.getPreviousSibling();
        }
        return result;
    }

    private static DetailAST getSecondOperand(DetailAST ast) {
        DetailAST result;
        if (ast.getLastChild() != null) {
            result = ast.getLastChild();
            while (result.getFirstChild() != null) {
                result = result.getFirstChild();
            }
        }
        else {
            result = ast.getNextSibling();
        }
        return result;
    }
```

It may be incomplete, there may be more nuances.
For now, it handles only one of them: `ASSIGN` in `VARIABLE_DEF` has only one child:
```
        int x = 0;
        x = x + 2;
```
The AST:
```
|--VARIABLE_DEF -> VARIABLE_DEF [5:8]
|   |--MODIFIERS -> MODIFIERS [5:8]
|   |--TYPE -> TYPE [5:8]
|   |   `--LITERAL_INT -> int [5:8]
|   |--IDENT -> x [5:12]
|   `--ASSIGN -> = [5:14]              // Left operand is previous sibling
|       `--EXPR -> EXPR [5:16]
|           `--NUM_INT -> 0 [5:16]
|--SEMI -> ; [5:17]
|--EXPR -> EXPR [6:10]
|   `--ASSIGN -> = [6:10]              // Left operand is first child
|       |--IDENT -> x [6:8]
|       `--PLUS -> + [6:14]
|           |--IDENT -> x [6:12]
|           `--NUM_INT -> 2 [6:16]
|--SEMI -> ; [6:17]
```
Please use this code as an example and fix all cases discovered by regression.

Reply:
Hey @pbludov if we add the comment nodes there will be more complication in the check and the code u wrote is complicated than the code currently it will be great if we go forward with the code i wrote in the Pull Request. @rnveach your views about this